### PR TITLE
Update rust: 1.58.1 -> 1.60.0

### PR DESCRIPTION
**Pull request by sveitser**
Also updates all flake inputs including nixpkgs.

### CODE REVIEW

**Comment by nyospe** [comment]
FYI, there's a clippy lint that (incorrectly, I believe, because pub trait methods are considered public interface) triggers for jellyfish with 1.60.0 (and apparently going back to 1.56.0)

error: methods called `is_*` usually take `self` by reference or no `self`
  --> primitives/src/circuit/schnorr_dsa.rs:68:9
   |
68 |         &mut self,
   |         ^^^^^^^^^
   |
note: the lint level is defined here
  --> primitives/src/lib.rs:11:9
   |
11 | #![deny(warnings)]
   |         ^^^^^^^^
   = note: `#[deny(clippy::wrong_self_convention)]` implied by `#[deny(warnings)]`
   = help: consider choosing a less ambiguous name
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention


This actually seems to be triggered by the removal of `wrong_pub_self_convention`

File: wallet/src/ui.rs [hunk 1]
```diff
@@ -457,19 +457,13 @@ impl TransactionHistoryEntry {
...
             hash: {
-                entry
-                    .hash
-                    .map(|hash| {
-                        bincode::serialize(&hash)
+                entry.hash.and_then(|hash| {
+                    bincode::serialize(&hash).ok().and_then(|bytes| {
+                        TaggedBase64::new("HASH", &bytes)
                             .ok()
-                            .map(|bytes| {
```
Comment (nyospe) [inline wallet/src/ui.rs:465]: These aren't valid cases for `flat_map()`?

Comment (sveitser) [reply wallet/src/ui.rs:465]: Not sure. Clippy said

```
error: called `map(..).flatten()` on an `Option`
   --> wallet/src/ui.rs:461:26
    |
461 |                       .hash
    |  __________________________^
462 | |                     .map(|hash| {
463 | |                         bincode::serialize(&hash)
464 | |                             .ok()
...   |
471 | |                     })
472 | |                     .flatten()
    | |______________________________^
    |
    = note: `-D clippy::map-flatten` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten
help: try using `and_then` instead
    |
461 ~                     .hash.and_then(|hash| {
462 +                         bincode::serialize(&hash)
463 +                             .ok()
464 +                             .map(|bytes| {
465 +                                 TaggedBase64::new("HASH", &bytes)
466 +                                     .ok()
  ...

error: called `map(..).flatten()` on an `Option`
   --> wallet/src/ui.rs:464:34
    |
464 |   ...                   .ok()
    |  ____________________________^
465 | | ...                   .map(|bytes| {
466 | | ...                       TaggedBase64::new("HASH", &bytes)
467 | | ...                           .ok()
468 | | ...                           .map(|tb| tb.to_string())
469 | | ...                   })
470 | | ...                   .flatten()
    | |________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten
help: try using `and_then` instead
    |
464 ~                             .ok().and_then(|bytes| {
465 +                                 TaggedBase64::new("HASH", &bytes)
466 +                                     .ok()
467 +                                     .map(|tb| tb.to_string())
468 +                             })
    |

error: could not compile `cape_wallet` due to 2 previous errors
```

Comment (sveitser) [reply wallet/src/ui.rs:465]: https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten

Comment (nyospe) [reply wallet/src/ui.rs:465]: Oh. Huh, yeah, I see.

**Review by nyospe — APPROVED** [review]

**Pull request merged**